### ISSUE TITLE
Connector builder cloud: rectify missing tests

### ISSUE THREAD

**Issue by sherifnada** [airbytehq/airbyte#25826]
### Revelant information

We recently had a problem where some dependencies weren't pinned in the connector builder package, and so when we deployed the platform, the connector builder image was rebuilt with a dependency which broke the application. 

This is very surprising because if the build succeeds then such a basic error shouldn't be possible. However it seems like currently the platform cloud deploy [builds images then deploys them](https://github.com/airbytehq/airbyte-platform-internal/actions/workflows/cloud-master-build.yml) (in the `Build and Push Images` step) without running E2E tests against them. 

This is problematic in the case that the Docker image being built has behavior which changes across time e.g: when deps aren't pinned or when it `RUN`s something 

This issue is suggesting that for the connector builder we should only deploy images which have already run against and passed CI. 

### Goals
1. Make sure we're running acceptance tests against the deployed images (counteract the negative symptoms discussed above)
2. Make sure we're testing auth logic in the connector builder against Airbyte Cloud

**Comment by mfsiega-airbyte** [comment]
tl;dr of my comments on slack - rather than changing what gets built/deployed, probably what you want is to make sure that the tests you want get run as part of the deploy process.

**Comment by sherifnada** [comment]
notes: 
1. Ideally we have an action which deploys cloud locally and runs our E2E test against it (would this also be useful for the platform?) as part of the PR builds
2. Our tests should also run against the deployed images on staging
3.
